backend/src/cache/storage/execution_cache_store_test.go

```diff
@@ -128,23 +129,41 @@ func TestGetExecutionCacheWithLatestCacheEntry(t *testing.T) {
 		EndedAtInSec:      2,
 	}
 	var executionCache *model.ExecutionCache
-	executionCache, err := executionCacheStore.GetExecutionCache("testKey")
+	executionCache, err := executionCacheStore.GetExecutionCache("testKey", -1)
 	require.Nil(t, err)
 	require.Equal(t, &executionCacheExpected, executionCache)
 }
 
+func TestGetExecutionCacheWithExpiredMaxCacheStaleness(t *testing.T) {
+	db := NewFakeDbOrFatal()
+	defer db.Close()
+	executionCacheStore := NewExecutionCacheStore(db, util.NewFakeTimeForEpoch())
+	executionCacheToPersist := &model.ExecutionCache{
+		ExecutionCacheKey: "test",
```

Comment:
Is the key correct?

Reply:
Ran the test case and package tests and it's correct. We are only testing GetExecutionCache() by creating a dummy cache entry first. The cache_key generation logic is not part of this class so cache_key here can be any string we want.

Reply:
Yes. But the key is "test" here and "testKey" later. Is that intentional?

Reply:
Oh that needs to be fixed. Thanks!